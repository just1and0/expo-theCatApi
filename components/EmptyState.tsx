import en from '@/constants/en';
import { View, Text, StyleSheet, Pressable } from 'react-native';

export function EmptyState({ onUpload }: { onUpload: () => void }) {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>{en.EMPTY_STATE.TITLE}</Text>
      <Text style={styles.message}>{en.EMPTY_STATE.SUBTITLE}</Text>
      <Pressable onPress={onUpload} style={styles.button}>
        <Text style={styles.buttonText}>{en.EMPTY_STATE.BUTTON}</Text>
      </Pressable>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 20,
    backgroundColor: '#fff',
  }, 
  title: {
    fontSize: 22,
    fontWeight: 'bold',
    color: '#333',
    marginBottom: 8,
  },
  message: {
    fontSize: 16,
    color: '#777',
    textAlign: 'center',
    marginBottom: 16,
  },
  button: {
    paddingVertical: 12,
    paddingHorizontal: 24,
    backgroundColor: '#ff7f50',
    borderRadius: 5,
  },
  buttonText: {
    color: '#fff',
    fontSize: 16,
    fontWeight: 'bold',
  },
});
