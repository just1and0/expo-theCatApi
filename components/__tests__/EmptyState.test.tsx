import * as React from 'react';
import renderer from 'react-test-renderer';

import { EmptyState } from '../EmptyState';

it(`renders correctly`, () => {
  const tree = renderer.create(
    <EmptyState
      onUpload={() => null} />).toJSON();

  expect(tree).toMatchSnapshot();
});
