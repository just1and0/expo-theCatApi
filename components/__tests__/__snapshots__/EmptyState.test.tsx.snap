// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "#333",
        "fontSize": 22,
        "fontWeight": "bold",
        "marginBottom": 8,
      }
    }
  >
    No Cats Available
  </Text>
  <Text
    style={
      {
        "color": "#777",
        "fontSize": 16,
        "marginBottom": 16,
        "textAlign": "center",
      }
    }
  >
    Looks like there are no cats to display right now. Upload a new cat image to get started!
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#ff7f50",
        "borderRadius": 5,
        "paddingHorizontal": 24,
        "paddingVertical": 12,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Upload Image
    </Text>
  </View>
</View>
`;
